src/element/lob.rs

```diff
@@ -0,0 +1,116 @@
+use crate::element::Bytes;
+
+/// An in-memory representation of an Ion blob.
+///
+/// ```rust
+/// use ion_rs::element::Blob;
+/// let ivm: &[u8] = &[0xEA_u8, 0x01, 0x00, 0xE0]; // Ion 1.0 version marker
+/// let blob: Blob = ivm.into();
+/// assert_eq!(&blob, ivm);
+/// assert_eq!(blob.as_slice().len(), 4);
+/// ```
+/// ```rust
+/// use ion_rs::element::Blob;
+/// let blob: Blob = "hello".into();
+/// assert_eq!(&blob, "hello".as_bytes());
+/// /// assert_eq!(blob.as_slice().len(), 5);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Blob(pub Bytes);
+
+impl Blob {
+    pub fn as_slice(&self) -> &[u8] {
+        self.as_ref()
+    }
+}
+
+/// An in-memory representation of an Ion clob.
+///
+/// ```rust
+/// use ion_rs::element::Clob;
+/// let clob: Clob = "hello".into();
+/// assert_eq!(&clob, "hello".as_bytes());
+/// assert_eq!(clob.as_slice().len(), 5);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Clob(pub Bytes);
+
+impl Clob {
+    pub fn as_slice(&self) -> &[u8] {
+        self.as_ref()
+    }
+}
+
+impl AsRef<[u8]> for Blob {
```

Comment:
🗺️ There are a lot of trait impls here that feel repetitive. I'm confident that we can represent them more concisely if needed (perhaps via a macro or a `Lob` trait with blanket implementations), but I want to leave that for a future PR.